src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java

```diff
@@ -149,4 +157,31 @@ private List<HashMap<String, Object>> getUpdateFieldValues(WfStatusEntityV2 stat
         }
         return updateFieldValuesList;
     }
+
+    private StringBuilder constructReadTermApiURI(String term, String category) {
+        StringBuilder builder = null;
+        if (!StringUtils.isEmpty(term) && !StringUtils.isEmpty(category)) {
+            builder = new StringBuilder();
+            UriComponents uriComponents = UriComponentsBuilder.fromUriString(host + termReadURI.replace(Constants.ID, term))
+                    .queryParam(Constants.FRAMEWORK, frameworkId).queryParam(Constants.CATEGORY, category)
+                    .build();
+            builder.append(uriComponents);
+        }
+        return builder;
+    }
+
+    private Map<String, Object> readTermObject(String term, String category) {
+        StringBuilder builder = null;
+        Map<String, Object> termResponse = null;
+        if (!StringUtils.isEmpty(term) && !StringUtils.isEmpty(category)) {
+            builder = new StringBuilder();
+            UriComponents uriComponents = UriComponentsBuilder.fromUriString(host + termReadURI.replace(Constants.ID, term))
```

Comment:
UriComponentBuilder is unnecessary -- you can directly use the StringBuilder instead.

Reply:
yes i did changes according to your suggestion please review.